rfcs/text/0002-type-ann-for-struct-properties-in-structs.md

```diff
@@ -0,0 +1,97 @@
+- Feature Name: type-annotations-for-struct-properties-in-structs
+- Start Date: TODO
+- RFC PR: (leave this empty)
+- Feature Commit(s): (leave this empty)
+
+# Summary
+
+Typecheck a value expression for a struct property in a struct definition with
+expected type information in the property.
+
+# Motivation
+By adding types for struct properties, we further complete the type annotations
+for struct types.
+
+# Guide-level explanation
+struct type properties are type-annotated with `Struct-Property`, where the only
+argument is the expected type of property value expressions. Here is an example
+of `prop:input-port`:
+
```

Comment:
To type-annotate a struct property we use the `Struct-Property`
form, which takes a single argument to describe the type of
that property's values. For example, here is a type-annotation
for the `prop:input-port` struct property: